### DISCUSSION TITLE
Changing the initialization point for TDS

### DISCUSSION THREAD

**Discussion by PandaRP01** [CURENT/andes#402] [Q&A]
Hello Mr. Cui and other experts,

We are simulating a distribution system with a number of batteries and solar PVs. We get decent results going through the normal process of initializing the system and running a dynamic simulation. However, we want to stop the simulation and save the latest values of all  system variable and restart the simulation by initializing the system variables using the saved values from the previous simulation. Could you please suggest a method to do so? 

We have also gone through the ANDES documentation and found that, if `init_tds=0`, all the variables are initialized after PFlow. If `init_tds=1`, variables can be initialized through files. If possible, please suggest the command 1) to store the latest values of variables during a simulation and 2) to use the save values to initialize in the next simulation.

Thank You and Regards
Rakesh Panda

**Comment by cuihantao** [comment]
The use case can be a bit tricky. First, let me make sure we are on the same page:

Initializing a new TDS study by results from a previous simulation does not guarantee a steady state initial condition. That is, this study will likely deviate from the given values starting from time zero. If done properly, this trajectory should be exactly the same as continuing the previous simulation from which you pulled data. Is this still what you are looking to do?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: PandaRP01 ***@***.***>
Sent: Friday, January 6, 2023 9:07:04 AM
To: cuihantao/andes ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [cuihantao/andes] Changing the initialization point for TDS (Discussion #402)


Hello Mr. Cui and other experts,

We are simulating a distribution system with a number of batteries and solar PVs. We get decent results going through the normal process of initializing the system and running a dynamic simulation. However, we want to stop the simulation and save the latest values of all system variable and restart the simulation by initializing the system variables using the saved values from the previous simulation. Could you please suggest a method to do so?

We have also gone through the ANDES documentation and found that, if init_tds=0, all the variables are initialized after PFlow. If init_tds=1, variables can be initialized through files. If possible, please suggest the command 1) to store the latest values of variables during a simulation and 2) to use the save values to initialize in the next simulation.

Thank You and Regards
Rakesh Panda

**Reply by PandaRP01** [reply]
Hello Mr. Cui

You are absolutely correct. We are trying to initialize the new `TDS` run from where the previous simulation ended. We are also expecting the trajectory of simulation to continue from exactly same point that the previous simulation ended. Kindly, suggest the appropriate method to do the initialization properly.

Thank You and Regards
Rakesh Panda

**Reply by cuihantao** [reply]
Hi Rakesh,

Currently, it is not possible to initialize a new system to a previous stopped one. The reason is that there are discontinuity states that are not externalized. 

Are you aware of the snapshot feature? See here: https://docs.andes.app/en/latest/_generated/andes.utils.snapshot.html. You can save the previous system as a snapshot and keep reloading it. It achieves the functionality of starting from where you stopped. The only difference is that time continues from where it stopped, not starting over from 0.

**Reply by PandaRP01** [reply]
Hello Mr. Cui

Thank you so much for the response. I don’t have much idea on this Snapshot feature. I followed the document and it is working in our system. It partially solves our issues and I will work further as per our objectives.

I am eagerly waiting for the upcoming release of ANDES where this initialization feature will be active. 
Thanks again for the great help. 

Regards 
Rakesh Panda

**Reply by cuihantao** [reply]
Thank you! I will put it in the release notes once it gets improved!
